src/closure-externs/webgpu-externs.js

```diff
@@ -1,42 +1,573 @@
+/** @return {undefined} */
+GPURenderBundleEncoder.prototype.pushDebugGroup = function() {};
+/** @return {undefined} */
+GPURenderBundleEncoder.prototype.popDebugGroup = function() {};
+/** @return {undefined} */
+GPURenderBundleEncoder.prototype.insertDebugMarker = function() {};
+/** @return {undefined} */
+GPURenderBundleEncoder.prototype.setBindGroup = function() {};
+/** @return {undefined} */
+GPURenderBundleEncoder.prototype.setBindGroup = function() {};
+/** @return {undefined} */
+GPURenderBundleEncoder.prototype.setPipeline = function() {};
+/** @return {undefined} */
+GPURenderBundleEncoder.prototype.setIndexBuffer = function() {};
+/** @return {undefined} */
+GPURenderBundleEncoder.prototype.setVertexBuffer = function() {};
+/** @return {undefined} */
+GPURenderBundleEncoder.prototype.draw = function() {};
+/** @return {undefined} */
+GPURenderBundleEncoder.prototype.drawIndexed = function() {};
+/** @return {undefined} */
+GPURenderBundleEncoder.prototype.drawIndirect = function() {};
+/** @return {undefined} */
+GPURenderBundleEncoder.prototype.drawIndexedIndirect = function() {};
+/** @return {!GPURenderBundle} */
+GPURenderBundleEncoder.prototype.finish = function() {};
+
+/** @constructor */
+function GPUQueue() {}
+/** @type {string} */
+GPUQueue.prototype.label;
+/** @return {undefined} */
+GPUQueue.prototype.submit = function() {};
+/** @return {!Promise<undefined>} */
+GPUQueue.prototype.onSubmittedWorkDone = function() {};
+/** @return {undefined} */
+GPUQueue.prototype.writeBuffer = function() {};
+/** @return {undefined} */
+GPUQueue.prototype.writeTexture = function() {};
+/** @return {undefined} */
+GPUQueue.prototype.copyExternalImageToTexture = function() {};
+
+/** @constructor */
+function GPUQuerySet() {}
+/** @type {string} */
+GPUQuerySet.prototype.label;
+/** @return {undefined} */
+GPUQuerySet.prototype.destroy = function() {};
+/** @type {string} */
+GPUQuerySet.prototype.type;
+/** @type {number} */
+GPUQuerySet.prototype.count;
+
+/** @constructor */
+function GPUCanvasContext() {}
+/** @type {!HTMLCanvasElement|!OffscreenCanvas} */
+GPUCanvasContext.prototype.canvas;
+/** @return {undefined} */
+GPUCanvasContext.prototype.configure = function() {};
+/** @return {undefined} */
+GPUCanvasContext.prototype.unconfigure = function() {};
+/** @return {!GPUTexture} */
+GPUCanvasContext.prototype.getCurrentTexture = function() {};
+
+/** @constructor */
+function GPUDeviceLostInfo() {}
+/** @type {string} */
+GPUDeviceLostInfo.prototype.reason;
+/** @type {string} */
+GPUDeviceLostInfo.prototype.message;
+
+/** @constructor */
+function GPUError() {}
+/** @type {string} */
+GPUError.prototype.message;
+
+/** @constructor */
+function GPUValidationError() {}
+
+/** @constructor */
+function GPUOutOfMemoryError() {}
+
+/** @constructor */
+function GPUInternalError() {}
+
+/** @constructor */
+function GPUUncapturedErrorEvent() {}
+/** @type {!GPUError} */
+GPUUncapturedErrorEvent.prototype.error;
```

Comment:
I wonder what closure compiler is doing about this upstream?  Might be worth contributing this there too.

Reply:
Unfortunately these types are pretty low-quality for upstream, as they don't define any function arguments, dictionaries, etc. But they should be sufficient for minification to work correctly so maybe they would want them anyway. I've filed https://github.com/google/closure-compiler/issues/4145